lib/ood_core/app_config.rb

```diff
@@ -0,0 +1,168 @@
+require "ood_core/refinements/array_extensions"
+require 'pathname'
+require 'yaml'
+
+module OodCore
+    class AppConfig
+        using Refinements::ArrayExtensions
+
+        attr_reader :raw_config
+
+        def initialize
+        end
+
+        def load_config
+            @raw_config = YAML.load(load_yaml) || {}
+        end
```

Comment:
`load_config` and `load_yaml` etc. should probably all be moved/grouped together as factory methods (as class methods or in a separate AppConfigFactory) and `initialize(raw_config)` so that to create the base config file you just do this: AppConfig.new({})

That could also help simplify some tests.

Reply:
It isn't clear to me yet how app specific ConfigurationSingleton classes will work with this "base class" (subclass? decorator? possibly decorator) but if individual apps are provided the option to instantiate the class (by calling, for example, AppConfig.load_config to get the raw config hash) then apps could load the base config hash, load the config the previous way, do a merge, and instantiate the resulting Configuration singleton, providing:

1. backwards compatibility with older configuration options
2. preservation of app specific configuration that does not belong here (like dataroot, database_path, etc.)

Reply:
I think that merging Hashes is the way to go; how else would we want to do it when we get to supporting user defined configuration?